### PR TITLE
Implement new style of LSTM function

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented new style of LSTM function. Its gradient implementation is based on old style Function, but we can replace it with a new style Function.

### CODE REVIEW

File: chainer/functions/activation/lstm.py [hunk 1]
```diff
@@ -117,20 +135,25 @@ def backward(self, inputs, grad_outputs):
         if gh is None:
             gh = 0
 
+        a, i, f, o = _extract_gates(x)
         if xp is numpy:
-            co = numpy.tanh(self.c)
+            tanh_a = xp.tanh(a)
+            sig_i = _sigmoid(i)
+            sig_f = _sigmoid(f)
+            sig_o = _sigmoid(o)
```
Comment (beam2d) [inline chainer/functions/activation/lstm.py]: It seems these intermediate values are still stored in the original `LSTM` object. Are they needed? Or is it better to reuse them as before (e.g. by passing them to `__init__`)?

Comment (unnonouno) [reply chainer/functions/activation/lstm.py]: Ah, ok. I think it is enough to remove `self.a`. How do you think?

Comment (beam2d) [reply chainer/functions/activation/lstm.py]: It's ok to remove them.

**Comment by beam2d** [comment]
Jenkins, test this please.

**Comment by beam2d** [comment]
Jenkins, test this please

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**